### PR TITLE
Dark mode toggle

### PR DESCRIPTION
**Pull request by jessmartin**
# Description

Toggle between dark mode and light mode.

## Type of change

- [x] New feature (non-breaking change that adds functionality)

## Screenshots/Screencaps

https://user-images.githubusercontent.com/27258/186490796-26fa9084-f9db-4e78-bb3b-a668b1ac5796.mov

### CODE REVIEW

File: src/lib/theme/index.ts [hunk 1]
```diff
@@ -4,7 +4,12 @@ export type Theme = 'light' | 'dark'
 
 export const loadTheme = (): Theme => {
   if (browser) {
-    return (localStorage.getItem('theme') as Theme) ?? 'light'
+    const browserTheme = localStorage.getItem('theme') as Theme
+    const osTheme = window.matchMedia('(prefers-color-scheme: dark)').matches
+      ? 'dark'
+      : 'light'
```
Comment (jessmartin) [inline src/lib/theme/index.ts:10]: This was pretty nifty!

File: src/lib/theme/index.ts [hunk 2]
```diff
@@ -4,7 +4,12 @@ export type Theme = 'light' | 'dark'
...
+
+    return browserTheme ?? (osTheme as Theme) ?? 'light'
```
Comment (jessmartin) [inline src/lib/theme/index.ts:12]: Nullish coalescing! 😄

Comment (bgins) [reply src/lib/theme/index.ts:12]: ❔❔ 🙌 🙌  🎉

File: src/lib/theme/index.ts (hunk 1 above)
Comment (bgins) [reply src/lib/theme/index.ts:10]: Yeah, that's awesome!

File: src/components/Header.svelte [hunk 3]
```diff
@@ -1,8 +1,16 @@
 <script lang="ts">
   import { goto } from '$app/navigation'
-  import { sessionStore } from '../stores'
+  import { sessionStore, theme } from '../stores'
+  import { Theme, storeTheme } from '$lib/theme/index'
```
Comment (bgins) [inline src/components/Header.svelte:4]: Could we import `Theme` as a type to make that explicit?

```suggestion
  import { storeTheme, type Theme } from '$lib/theme/index'
```

**Review by bgins — APPROVED** [review]
Looks good, love how the button turned out and great to use the OS theme!

**Pull request merged**